From fe21763c229753f1313e9e3f44b10563335c921d Mon Sep 17 00:00:00 2001
From: Baokun Li <libaokun1@huawei.com>
Date: Tue, 13 Sep 2022 23:00:21 +0800
Subject: [PATCH 0186/1076] [Huawei] dm ioctl: print error when HC and MD do
 not match

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187327

--------------------------------

Print error when hc and md do not match, which is convenient for locating
the cause of the problem

Signed-off-by: Baokun Li <libaokun1@huawei.com>
---
 drivers/md/dm-ioctl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/dm-ioctl.c b/drivers/md/dm-ioctl.c
index ace4301b3a49..2186a3a4e48b 100644
--- a/drivers/md/dm-ioctl.c
+++ b/drivers/md/dm-ioctl.c
@@ -2040,6 +2040,8 @@ int dm_copy_name_and_uuid(struct mapped_device *md, char *name, char *uuid)
 	mutex_lock(&dm_hash_cells_mutex);
 	hc = dm_get_mdptr(md);
 	if (!hc || hc->md != md) {
+		if (hc)
+			DMERR("hash cell and mapped device do not match!");
 		r = -ENXIO;
 		goto out;
 	}
-- 
2.17.1

